### PR TITLE
Update external-oauth.md

### PR DESCRIPTION
**Pull request by runleonarun**
Add some Troubleshooting guidance for an external Oauth specific error

## What are you changing in this pull request and why?

Content added by @gunnava! Thanks so much!

## Checklist
- [ ] I have reviewed the [Content style guide](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md) so my content adheres to these guidelines.
- [ ] The topic I'm writing about is for specific dbt version(s) and I have versioned it according to the [version a whole page](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#adding-a-new-version) and/or [version a block of content](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#versioning-blocks-of-content) guidelines.
- [ ] I have added checklist item(s) to this list for anything anything that needs to happen before this PR is merged, such as "needs technical review" or "change base branch."
- [ ] The content in this PR requires a dbt release note, so I added one to the [release notes page](https://docs.getdbt.com/docs/dbt-versions/dbt-cloud-release-notes).

### CODE REVIEW

File: website/docs/faqs/Troubleshooting/failed-snowflake-oauth-connection.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+---
+title: Receiving a `Failed to connect to DB xxxxxxx.snowflakecomputing.com:443` error when when connecting to Snowflake via external Oauth
+description: "Edit your OAuth Security integration when you see error"
+sidebar_label: 'Receiving `Failed to connect to database` error'
+---
+
+If you see the following error w
```
Comment (runleonarun) [inline website/docs/faqs/Troubleshooting/failed-snowflake-oauth-connection.md]:
```suggestion
1. If you see this error: 
```

File: website/docs/faqs/Troubleshooting/failed-snowflake-oauth-connection.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+```
+Failed to connect to DB: xxxxxxx.snowflakecomputing.com:443. The role requested in the connection or the default role if none was requested in the connection ('xxxxx') is not listed in the Access Token or was filtered. Please specify another role, or contact your OAuth Authorization server administrator.
+```
+
+To resolve this issue, you should edit your OAuth Security integration and explicitly specify this Scope mapping attribute:
```
Comment (runleonarun) [inline website/docs/faqs/Troubleshooting/failed-snowflake-oauth-connection.md]:
```suggestion
2. Edit your OAuth Security integration and explicitly specify this Scope mapping attribute:
```

File: website/docs/faqs/Troubleshooting/failed-snowflake-oauth-connection.md [hunk 3]
```diff
@@ -0,0 +1,18 @@
+---
+title: Receiving a `Failed to connect to DB xxxxxxx.snowflakecomputing.com:443` error when when connecting to Snowflake via external Oauth
```
Comment (runleonarun) [inline website/docs/faqs/Troubleshooting/failed-snowflake-oauth-connection.md]:
```suggestion
title: Receiving a `Failed to connect to DB` error when connecting to Snowflake
```

File: website/docs/faqs/Troubleshooting/failed-snowflake-oauth-connection.md [hunk 4]
```diff
@@ -0,0 +1,18 @@
+---
+title: Receiving a `Failed to connect to DB` error when connecting to Snowflake
+description: "Edit your OAuth Security integration when you see error"
+sidebar_label: 'Receiving `Failed to connect to database` error'
+---
+
+1. If you see this error: 
+
+```
```
Comment (runleonarun) [inline website/docs/faqs/Troubleshooting/failed-snowflake-oauth-connection.md]:
```suggestion
```shell
```

File: website/docs/faqs/Troubleshooting/failed-snowflake-oauth-connection.md [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+```shell
+Failed to connect to DB: xxxxxxx.snowflakecomputing.com:443. The role requested in the connection or the default role if none was requested in the connection ('xxxxx') is not listed in the Access Token or was filtered. Please specify another role, or contact your OAuth Authorization server administrator.
+```
+
+2. Edit your OAuth Security integration and explicitly specify this Scope mapping attribute:
+
+ ```sql
+ALTER INTEGRATION <my_int_name> SET EXTERNAL_OAUTH_SCOPE_MAPPING_ATTRIBUTE = 'scp';
+```
+You can read more about this error in [Snowflake's documentation](https://community.snowflake.com/s/article/external-custom-oauth-error-the-role-requested-in-the-connection-is-not-listed-in-the-access-token).
```
Comment (runleonarun) [inline website/docs/faqs/Troubleshooting/failed-snowflake-oauth-connection.md:22]:
```suggestion

You can read more about this error in [Snowflake's documentation](https://community.snowflake.com/s/article/external-custom-oauth-error-the-role-requested-in-the-connection-is-not-listed-in-the-access-token).
```

File: website/docs/faqs/Troubleshooting/failed-snowflake-oauth-connection.md [hunk 6]
```diff
@@ -0,0 +1,19 @@
+---
+title: Receiving a `Failed to connect to DB` error when connecting to Snowflake
+description: "Edit your OAuth Security integration when you see error"
+sidebar_label: 'Receiving `Failed to connect to database` error'
+---
+
+1. If you see this error: 
+
+```shell
+Failed to connect to DB: xxxxxxx.snowflakecomputing.com:443. The role requested in the connection or the default role if none was requested in the connection ('xxxxx') is not listed in the Access Token or was filtered. Please specify another role, or contact your OAuth Authorization server administrator.
```
Comment (runleonarun) [inline website/docs/faqs/Troubleshooting/failed-snowflake-oauth-connection.md]:
```suggestion
Failed to connect to DB: xxxxxxx.snowflakecomputing.com:443. The role requested in the connection or the default role if none was requested in the connection ('xxxxx') is not listed in the Access Token or was filtered. 
Please specify another role, or contact your OAuth Authorization server administrator.
```

File: website/docs/faqs/Troubleshooting/failed-snowflake-oauth-connection.md [hunk 7]
```diff
@@ -0,0 +1,20 @@
+---
+title: Receiving a `Failed to connect to DB` error when connecting to Snowflake
+description: "Edit your OAuth Security integration when you see error"
+sidebar_label: 'Receiving `Failed to connect to database` error'
+---
+
+1. If you see this error: 
+
+```shell
+Failed to connect to DB: xxxxxxx.snowflakecomputing.com:443. The role requested in the connection or the default role if none was requested in the connection ('xxxxx') is not listed in the Access Token or was filtered.
```
Comment (runleonarun) [inline website/docs/faqs/Troubleshooting/failed-snowflake-oauth-connection.md]:
```suggestion

Failed to connect to DB: xxxxxxx.snowflakecomputing.com:443. The role requested in the connection or the default role if none was requested in the connection ('xxxxx') is not listed in the Access Token or was filtered. 
```

File: website/docs/faqs/Troubleshooting/failed-snowflake-oauth-connection.md [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+sidebar_label: 'Receiving `Failed to connect to database` error'
+---
+
+1. If you see this error: 
+
+```shell
+
+Failed to connect to DB: xxxxxxx.snowflakecomputing.com:443. The role requested in the connection or the default role if none was requested in the connection ('xxxxx') is not listed in the Access Token or was filtered. 
+Please specify another role, or contact your OAuth Authorization server administrator.
+```
```
Comment (runleonarun) [inline website/docs/faqs/Troubleshooting/failed-snowflake-oauth-connection.md]:
```suggestion

```
```

File: website/docs/faqs/Troubleshooting/failed-snowflake-oauth-connection.md [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+description: "Edit your OAuth Security integration when you see error"
+sidebar_label: 'Receiving `Failed to connect to database` error'
+---
+
+1. If you see this error: 
+
+```shell
+
+Failed to connect to DB: xxxxxxx.snowflakecomputing.com:443. The role requested in the connection or the default role if none was requested in the connection ('xxxxx') is not listed in the Access Token or was filtered. 
+Please specify another role, or contact your OAuth Authorization server administrator.
```
Comment (runleonarun) [inline website/docs/faqs/Troubleshooting/failed-snowflake-oauth-connection.md]:
```suggestion
Please specify another role, or contact your OAuth Authorization server administrator.

```
```

File: website/docs/faqs/Troubleshooting/failed-snowflake-oauth-connection.md [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+title: Receiving a `Failed to connect to DB` error when connecting to Snowflake
+description: "Edit your OAuth Security integration when you see error"
+sidebar_label: 'Receiving `Failed to connect to database` error'
+---
+
+1. If you see this error: 
+
+   ```shell
+   
+   Failed to connect to DB: xxxxxxx.snowflakecomputing.com:443. The role requested in the connection or the default role if none was requested in the connection ('xxxxx') is not listed in the Access Token or was filtered.
```
Comment (nataliefiann) [inline website/docs/faqs/Troubleshooting/failed-snowflake-oauth-connection.md]:
```suggestion
   Failed to connect to DB: xxxxxxx.snowflakecomputing.com:443. The role requested in the connection, or the default role if none was requested in the connection ('xxxxx'), is not listed in the Access Token or was filtered. 
```

File: website/docs/faqs/Troubleshooting/failed-snowflake-oauth-connection.md [hunk 11]
```diff
@@ -0,0 +1,22 @@
...
+1. If you see this error: 
+
+   ```shell
+   
+   Failed to connect to DB: xxxxxxx.snowflakecomputing.com:443. The role requested in the connection or the default role if none was requested in the connection ('xxxxx') is not listed in the Access Token or was filtered. 
+   Please specify another role, or contact your OAuth Authorization server administrator.
+   
+   ```
+
+2. Edit your OAuth Security integration and explicitly specify this Scope mapping attribute:
```
Comment (nataliefiann) [inline website/docs/faqs/Troubleshooting/failed-snowflake-oauth-connection.md]:
```suggestion
2. Edit your OAuth Security integration and explicitly specify this scope mapping attribute:
```

Comment (nataliefiann) [reply website/docs/faqs/Troubleshooting/failed-snowflake-oauth-connection.md]: Hiya @runleonarun 

I've checked the snowflake docs and I'm not 100% sure if the "s" in Scope mapping attribute should be capitalised or not. In some of the Snowflake docs, the "s" is capitalised and in others, it isn't.

Kind Regards
Natalie

**Review by nataliefiann — APPROVED** [review]
Hiya @runleonarun 

Thank you for creating this! I've approved this for you. I've added two non-block suggestions.

Kind Regards
Natalie

**Pull request merged**